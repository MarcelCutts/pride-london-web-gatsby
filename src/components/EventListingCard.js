import React, { Component } from 'react';
import PropTypes from 'prop-types'
import Link from 'gatsby-link'
import moment from 'moment'
import styled from 'styled-components'

// const Card = styled.div`
  
// `

export default class EventListingCard extends Component {
  formatDate = () => {
    const year = moment(this.props.event.startTime).year()

    const startDate = moment(this.props.event.startTime).format('L')
    const endDate = moment(this.props.event.endTime).format('L')

    const startMonth = moment(this.props.event.startTime).format('MMM')
    const endMonth = moment(this.props.event.endTime).format('MMM')

    const startDay = moment(this.props.event.startTime).date()
    const endDay = moment(this.props.event.endTime).date()

    const startTime = this.formatTime(this.props.event.startTime);
    const endTime = this.formatTime(this.props.event.endTime);

    if(startDate === endDate) {
      return `${startDay} ${startMonth} ${year} • ${startTime} - ${endTime}`
    }
    else if (startMonth === endMonth) {
      return `${startDay} - ${endDay} ${startMonth} ${year} • ${startTime} - ${endTime}`
    }
    else {
      return `${startDay} ${startMonth} - ${endDay} ${endMonth} ${year} • ${startTime} - ${endTime}`
    }
  }

  formatTime = (time) => {
    if(moment(time).format('mm') === '00') {
      return moment(time).format('ha')
    } 
    else {
      return moment(time).format('h:mma')
    }
  }

  render() {
    const event = this.props.event

    return (
      <div>
        <img 
            src={`${event.eventsListPicture.file.url}?fit=fill&w=400&h=225&f=faces`} 
            alt={event.eventsListPicture.title}
            width="400"
            height="225"
        />   
        <h2><Link to={`/events/${event.id}`}>{event.name}</Link></h2>
        <p>{this.formatDate()}</p>
      </div>
    )   
  }
}

EventListingCard.propTypes = {
  event: PropTypes.object
}