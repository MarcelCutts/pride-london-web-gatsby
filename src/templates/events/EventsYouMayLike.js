import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import moment from 'moment'
import EventListingCard from './EventListingCard'
import { Consumer } from '../../components/AppContext'
import { Container, Row, Column } from '../../components/grid/grid'

const ViewAll = styled.a`
  color: ${props => props.theme.colors.indigo};
  display: block;
  font-family: ${props => props.theme.fonts.title};
  font-size: 0.8em;
  letter-spacing: 0.1px;
  padding-top: 5px;
  text-align: right;
  text-decoration: none;
  width: 100%;
`

const FlexColumn = styled(Column)`
  display: flex;
`

const filterEventsYouMayLike = (events, eventId) => {
  const filteredEvents = events.filter(event => {
    if (event.node.id === eventId) return false

    return moment(event.node.startTime).diff(moment(), 'hours') > 0
  })

  return filteredEvents.splice(0, 3)
}

const EventsYouMayLike = ({ eventId }) => (
  <Consumer>
    {({ state: { events } }) => {
      const eventsYouMayLike = filterEventsYouMayLike(events, eventId)

      if (eventsYouMayLike.length === 0) return null

      return (
        <Container>
          <Row>
            <Column width={0.7}>
              <h2>You may also like</h2>
            </Column>
            <Column width={0.3}>
              <ViewAll href="/events">View all events {'>'}</ViewAll>
            </Column>
          </Row>
          <Row>
            {eventsYouMayLike.map(event => (
              <FlexColumn width={[1, 1 / 2, 1 / 3, 1 / 3]} key={event.node.id}>
                <EventListingCard event={event.node} />
              </FlexColumn>
            ))}
          </Row>
        </Container>
      )
    }}
  </Consumer>
)

EventsYouMayLike.propTypes = {
  eventId: PropTypes.string,
}

export default EventsYouMayLike
