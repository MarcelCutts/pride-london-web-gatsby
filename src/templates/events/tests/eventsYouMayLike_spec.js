import React from 'react'
import { mount } from 'enzyme'
import EventsYouMayLike from '../eventsYouMayLike'
import EventListingCard from '../eventListingCard'

beforeEach(() => {
  jest.resetModules()
})

jest.mock('../../../components/appContext', () => ({
  Consumer: props =>
    props.children({
      events: [
        {
          node: {
            id: '123',
            startTime: '2035-06-02T13:30+01:00',
          },
        },
      ],
    }),
}))

describe('The Events You May Like component', () => {
  const mockProps = {
    eventId: '1234',
  }

  it('should matchsnapshot', () => {
    const wrapper = mount(<EventsYouMayLike {...mockProps} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should render Event Listing Card if one event is passed from the state', () => {
    const wrapper = mount(<EventsYouMayLike {...mockProps} />)

    expect(wrapper.find(EventListingCard).length).toBe(1)
  })

  it('should not render an item that started in the past', () => {
    const wrapper = mount(<EventsYouMayLike {...mockProps} />)

    expect(wrapper.find(EventListingCard).length).toBe(0)
  })

  it('should not render more than 3 items', () => {
    const wrapper = mount(<EventsYouMayLike {...mockProps} />)

    expect(wrapper.find(EventListingCard).length).toBe(3)
  })

  it('should render an item that has the same id as the current event', () => {
    const wrapper = mount(<EventsYouMayLike {...mockProps} />)

    expect(wrapper.find(EventListingCard).length).toBe(0)
  })
})
