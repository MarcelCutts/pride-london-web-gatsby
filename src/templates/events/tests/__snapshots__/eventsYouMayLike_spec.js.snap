// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Events You May Like component should matchsnapshot 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <EventsYouMayLike
    eventId="1234"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "eventId": "1234",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "eventId": "1234",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
